lib/Byng/Pimcore/Sitemap/Generator/SitemapGenerator.php

```diff
@@ -65,18 +66,51 @@ public function __construct()
      */
     public function generateXml()
     {
-        // Get all the root elements with parentId '1'
-        $rootDocuments = $this->documentGateway->getChildren(1);
-
-        foreach ($rootDocuments as $rootDocument) {
-            $this->addUrlChild($rootDocument);
-            $this->listAllChildren($rootDocument);
+        // Retrieve site trees
+        $siteRoots = new Site\Listing();
+        $siteRoots = $siteRoots->load();
+
+        // Build siteRoots table: [ ID => Domain ]
+        /* @var Site $siteRoot */
+        foreach ($siteRoots as $siteRoot){
+            $this->sitesRoots[$siteRoot->getRootId()] = $siteRoot->getMainDomain();
         }
-        $this->xml->asXML(PIMCORE_DOCUMENT_ROOT . "/sitemap.xml");
 
-        if (Config::getSystemConfig()->get("general")->get("environment") === "production") {
-            $this->notifySearchEngines();
+        // Also append the default tree
+        $this->sitesRoots[1] = Config::getSystemConfig()->get("general")->get("domain");
+
+        $notifySearchEngines = Config::getSystemConfig()->get("general")->get("environment") === "production";
+        foreach($this->sitesRoots as $siteRootID => $siteRootDomain){
+            $this->generateSiteXml($siteRootID, $siteRootDomain);
+
+            if ($notifySearchEngines)
+                $this->notifySearchEngines('https://' . $siteRootDomain);
         }
+
+    }
+
+    /**
+     * Generate a sitemap xml file for a defined site, with a specific hostUrl
+     *
+     * @param $rootId
+     * @param $hostUrl
+     * @return string
+     */
+    private function generateSiteXml($rootId, $hostUrl)
+    {
+        $this->xml = new SimpleXMLElement(
+            '<?xml version="1.0" encoding="UTF-8"?>'
+            . '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"></urlset>'
+        );
+
+        // Set current hostUrl
+        $this->hostUrl = 'https://' . $hostUrl;
```

Comment:
I think a configuration file may be the best of the options there. Even with caching the curl result, Pimcore isn't always that speedy. I think that whilst website settings are indeed convenient this kind of configuration doesn't really need to be in the UI at all (keep it away from people who may potentially muck it up! :joy:).